mesh/ncmesh.cpp

```diff
@@ -487,6 +502,111 @@ int NCMesh::Face::GetSingleElement() const
 
 //// Refinement ////////////////////////////////////////////////////////////////
 
+void Refinement::SetScale(const ScaledType &r)
+{
+   switch (r.first)
+   {
+      case X:
+         s[0] = r.second;
+         break;
+      case Y:
+         s[1] = r.second;
+         break;
+      case Z:
+         s[2] = r.second;
+         break;
+      case XY:
+         s[0] = r.second;
+         s[1] = r.second;
+         break;
+      case YZ:
+         s[1] = r.second;
+         s[2] = r.second;
+         break;
+      case XZ:
+         s[0] = r.second;
+         s[2] = r.second;
+         break;
+      case XYZ:
+         s[0] = r.second;
+         s[1] = r.second;
+         s[2] = r.second;
+         break;
+      default:
+         MFEM_ABORT("Unsupported refinement type.");
+   }
+}
+
+Refinement::Refinement(int index)
+   : index(index)
+{
+   for (int i=0; i<3; ++i) { s[i] = 0.0; }
+   // Default case is XYZ type with scale 0.5.
+   SetScale(ScaledType(XYZ, 0.5));
+}
+
+Refinement::Refinement(int index, const std::initializer_list<ScaledType> &refs)
+   : index(index)
+{
+   for (int i=0; i<3; ++i) { s[i] = 0.0; }
+   if (refs.size() == 0)
+   {
+      // Default case is XYZ type with scale 0.5.
+      SetScale(ScaledType(XYZ, 0.5));
+   }
+   else
+   {
+      for (const auto & ref : refs)
+      {
+         SetScale(ref);
+      }
+   }
+}
+
+Refinement::Refinement(int index, char type, real_t scale)
+   : index(index)
+{
+   for (int i=0; i<3; ++i) { s[i] = 0.0; }
+   SetScale(ScaledType(type, scale));
+}
+
+Refinement::Refinement(int index, int type, real_t scale)
+   : index(index)
+{
+   for (int i=0; i<3; ++i) { s[i] = 0.0; }
+   SetScale(ScaledType((char) type, scale));
+}
+
+int Refinement::GetType() const
+{
+   int t = 0;
+   int d = 1;
+   for (int i=0; i<3; ++i)
+   {
+      if (s[i] > 0.0)
+      {
+         t += d;
+      }
+
+      d *= 2;
+   }
+
+   return t;
+}
+
+void Refinement::Set(int element, int type, real_t scale)
+{
+   index = element;
+   for (int i=0; i<3; ++i) { s[i] = 0.0; }
+   SetScale(ScaledType((char) type, scale));
+}
```

Comment:
Given there's no backwards compat issue, might as well just have `char type` in the arguments and not cast. If the int need to be kept, mark every method which uses it `MFEM_DEPRECATED` and use the char versions only internally to mfem.